src/main/java/dev/andrylat/vomelianchuk/card_number_operations/CardNumber.java

```diff
@@ -0,0 +1,39 @@
+package dev.andrylat.vomelianchuk.card_number_operations;
+
+import java.util.List;
+
+public class CardNumber {
```

Comment:
Now, everything is much more better. I like it.

but solution is still not production like.

According to SOLID, your class should have only one responsibility. Here I see two of them:
- run Java application (main())
- validate card.

It is not correct. This class should be responsible only for running Java application. What does it mean? It means that only running related things should be here. What are they? 
 - Work with arguments
 - Main application flow (next task will be about adding mortgage calculator and selecting correct UI should be here)
 - etc...
 
 My main ideas is about moving all related stuff with validation to a separate class.

Reply:
the good solution should be like:
```
main() {
  UserConsoleIO  userConsoleIO = new UserConsoleIO(System.in, System.out);
  var cardNumberAsString = userConsoleIO.readCardNumber();
  try {
    new CardNumbervalidator().validate(cardNumberAsString);
    PaymentSystem paymentSystem = PaymentSystem.of(cardNumberAsString);
    userConsoleIO.printPaymentSytem(paymentSystem);
  } catch ( CardValidationException | WrongPaymentSystemException ex) {
    userConsoleIO.printErrors(ex);
  }
}

Reply:
Fixed here: c79cac4